libs/platform/unit/platform_unit_test.cpp

```diff
@@ -440,6 +441,73 @@ test_provider_attach_client(
     return STATUS_SUCCESS;
 };
 
+static NTSTATUS
+test_denied_client_attach_provider(
+    HANDLE nmr_binding_handle,
+    _Inout_ void* client_context,
+    _In_ const NPI_REGISTRATION_INSTANCE* provider_registration_instance,
+    _In_ const void* provider_binding_context,
+    _In_ const void* provider_dispatch,
+    _Out_ void** client_binding_context,
+    _Out_ const void** client_dispatch)
+{
+    ebpf_extension_client_t* client = (ebpf_extension_client_t*)client_context;
+    UNREFERENCED_PARAMETER(nmr_binding_handle);
+    UNREFERENCED_PARAMETER(client);
+    provider_registration_instance = NULL;
+    UNREFERENCED_PARAMETER(provider_binding_context);
+    UNREFERENCED_PARAMETER(provider_dispatch);
+    *client_binding_context = nullptr;
+    *client_dispatch = &test_provider_dispatch_table;
+    test_denied_attach_called = true;
+    return STATUS_ACCESS_DENIED;
+};
+
+static NTSTATUS
+test_failed_provider_attach_client(
+    HANDLE nmr_binding_handle,
+    _Inout_ void* provider_context,
+    _In_ const NPI_REGISTRATION_INSTANCE* client_registration_instance,
+    _In_ const void* client_binding_context,
+    _In_ const void* client_dispatch,
+    _Out_ void** provider_binding_context,
+    _Out_ const void** provider_dispatch)
+{
+    ebpf_extension_provider_t* provider = (ebpf_extension_provider_t*)provider_context;
+    UNREFERENCED_PARAMETER(nmr_binding_handle);
+    UNREFERENCED_PARAMETER(provider);
+    UNREFERENCED_PARAMETER(client_registration_instance);
+    UNREFERENCED_PARAMETER(client_binding_context);
+    UNREFERENCED_PARAMETER(client_dispatch);
+    *provider_binding_context = nullptr;
+    *provider_dispatch = &test_provider_dispatch_table;
+    test_denied_attach_called = true;
+    return STATUS_ACCESS_DENIED;
+};
+
+static NTSTATUS
+test_failed_provider_register_client(
```

Comment:
This seems to be identical to the previous function. Did you mean to return a different failure code?